pkg/registry/blobs.go

```diff
@@ -76,6 +76,39 @@ type blobPutHandler interface {
 	Put(ctx context.Context, repo string, h v1.Hash, rc io.ReadCloser) error
 }
 
+// uploadHandler represents a minimal upload storage backend, capable of
+// appending streamed upload contents and finally returning them to be stored
+// in blob storage.
+type uploadHandler interface {
+	// StatUpload returns the current size of the given upload.
+	StatUpload(ctx context.Context, uploadID string) (int64, error)
+
+	// AppendUpload appends the contents of the ReadCloser to the current
+	// upload contents, and returns the new total size.
+	AppendUpload(ctx context.Context, uploadID string, rc io.ReadCloser) (int64, error)
```

Comment:
Can we [ab]use `io.WriterAt` to express this instead? Feels like it maps pretty well to chunked uploads.